frontend/src/pages/workspaces.js

```diff
@@ -0,0 +1,211 @@
+import React, { Component } from 'react'
+import { Container, Table } from 'reactstrap'
+import Nav from '../components/nav'
+import Head from '../components/head'
+import WorkspaceModal from '../components/workspaceModal'
+import {
+  getWorkspaces,
+  createWorkspace,
+  createCycle,
+  getCyclesByWorkspace,
+  setCurrentCycle,
+} from '../utils/api'
+import { newWorkspaceFields, newCycleFields } from '../utils/formFields'
+import { ActionButton } from '../components/common'
+
+class Workspaces extends Component {
+  constructor(props) {
+    super(props)
+    this.state = {
+      workspaces: [],
+      selectedWorkspace: '',
+      cycles: [],
+      workspaceModal: false,
+      cycleModal: false,
+    }
+  }
+
+  async componentDidMount() {
+    // get workspaces and cycle, if applicable
+    const workspaceRes = await getWorkspaces()
+    if (workspaceRes) {
+      workspaceRes.result.forEach(async workspace => {
```

Comment:
why is workspace async? and this could probably be more intuitive as the following:
```js
if (workspaceRes) {
  this.setState({ 
    workspaces: workspaceRes.result.map(workspace => workspace.name),
  });
}
```

Reply:
i don't remember why i made it async, but that suggestion seems to work